modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1832,51 +1817,19 @@ public CompletableFuture<List<Table>> tablesAsync() {
      * @return Future representing pending completion of the operation.
      */
     private CompletableFuture<List<Table>> tablesAsyncInternal() {
-        return supplyAsync(() -> inBusyLock(busyLock, this::directTableIds), ioExecutor)
-                .thenCompose(tableIds -> inBusyLock(busyLock, () -> {
-                    var tableFuts = new CompletableFuture[tableIds.size()];
-
-                    var i = 0;
-
-                    for (int tblId : tableIds) {
-                        tableFuts[i++] = tableAsyncInternal(tblId, false);
-                    }
-
-                    return allOf(tableFuts).thenApply(unused -> inBusyLock(busyLock, () -> {
-                        var tables = new ArrayList<Table>(tableIds.size());
-
-                        for (var fut : tableFuts) {
-                            var table = fut.join();
-
-                            if (table != null) {
-                                tables.add((Table) table);
-                            }
-                        }
-
-                        return tables;
-                    }));
-                }));
+        return schemaSyncService.waitForMetadataCompleteness(clock.now())
+                .thenApply(ignore -> List.copyOf(latestTablesById().values()));
```

Comment:
It looks like `tablesAsync()` works differently from `tableAsync(String)` in one aspect. The latter waits for the table that it sees as 'not yet ready' to become ready, so it will not return `null` right away for a non-ready table; but the formor will just return ready tables. So from the point of view of a user, this might look inconsistent: if there is just 1 table that is not ready yet, `tablesAsync()` will return an empty collection immediately after the metadata sync is over, but `tableAsync(String)` will wait and return the table once it's ready.

How about aligning the behaviors of the methods? Like making `tablesAsync()` wait till all tables are ready? I think this is how it worked before the changes proposed in this PR.